pkg/cmd/pulumi/about.go

```diff
@@ -0,0 +1,304 @@
+// Copyright 2016-2018, Pulumi Corporation.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"runtime"
+	"sort"
+
+	"github.com/pkg/errors"
+	"github.com/shirou/gopsutil/host"
+	"github.com/spf13/cobra"
+
+	"github.com/pulumi/pulumi/pkg/v3/backend"
+	"github.com/pulumi/pulumi/pkg/v3/backend/display"
+	"github.com/pulumi/pulumi/pkg/v3/backend/state"
+	"github.com/pulumi/pulumi/pkg/v3/resource/deploy"
+	"github.com/pulumi/pulumi/pkg/v3/version"
+	"github.com/pulumi/pulumi/sdk/v3/go/common/diag/colors"
+	"github.com/pulumi/pulumi/sdk/v3/go/common/resource"
+	"github.com/pulumi/pulumi/sdk/v3/go/common/util/cmdutil"
+	"github.com/pulumi/pulumi/sdk/v3/go/common/workspace"
+)
+
+func newAboutCmd() *cobra.Command {
+	short := "Print information about the pulumi enviroment."
+	cmd :=
+		&cobra.Command{
+			Use:   "about",
+			Short: short,
+			Long: short + "\n" +
+				"\n" +
+				"Prints out information helpful for debugging the pulumi CLI." +
+				"\n" +
+				"This includes information about:\n" +
+				" - the CLI and how it was built\n" +
+				" - which OS pulumi was run from\n" +
+				" - the current project\n" +
+				" - the current stack\n" +
+				" - the current backend\n",
+			Args: cmdutil.MaximumNArgs(0),
+			Run: cmdutil.RunFunc(func(cmd *cobra.Command, args []string) error {
+				var err error
+				opts := display.Options{
+					Color: cmdutil.GetGlobalColorization(),
+				}
+
+				// TODO Maybe include a message saying to include this info when
+				// starting an issue.
+
+				formatCLIAbout()
+				fmt.Print("\n")
+
+				if err = formatPluginAbout(); err != nil {
+					err = errors.Wrap(err, "Failed to get information about the plugin.")
+					warn(err, &opts)
+				}
+				fmt.Print("\n")
+
+				if err = formatHostAbout(); err != nil {
+					err = errors.Wrap(err, "Failed to get information about the host")
+					warn(err, &opts)
+				}
+				fmt.Print("\n")
+
+				var proj *workspace.Project
+				var pwd string
+				if proj, pwd, err = readProject(); err != nil {
+					err = errors.Wrap(err, "Failed to read project for diagnosis")
+					warn(err, &opts)
+				} else {
+					fmt.Printf("This is a %s project.\n\n", proj.Runtime.Name())
```

Comment:
Also would be nice to get runtime version

Reply:
The runtime version of what? Can you please clarify?

Reply:
Version of node/python/etc. Figure let's collect all the helpful telemetry while we're here